UDT.Business/Implementation/SubjectService.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using UDT.Business.Interfaces;
+using UDT.Model.Entities;
+using UDT.Repository;
+
+namespace UDT.Business.Implementation
+{
+    public class SubjectService : ISubjectService
+    {
+        private readonly DataContext _dataContext;
+
+        public SubjectService(DataContext dataContext)
+        {
+            _dataContext = dataContext;
+        }
+
+        public IAsyncEnumerable<Subject> GetAllAsync()
+        {
+            return _dataContext.Subjects
+                .AsAsyncEnumerable();
+        }
+
+        public async Task<Subject> GetByIdAsync(int id)
+        {
+            return await _dataContext.Subjects
+                .Include(subject => subject.Users)
```

Comment:
Nu cred ca ar trebui sa includem userii la o materie, doar in cazul in care este nevoie.
Adica problema e ca daca vrem doar sa afisam o materie, o sa umple obiectul cu toti cei 250 de studenti care participa